tools/build_defs/repo/utils.bzl

```diff
@@ -71,16 +71,16 @@ def _use_native_patch(patch_args):
     return True
 
 def _download_patch(ctx, patch_url, integrity, auth = None):
-    name = patch_url.split("/")[-1]
-    patch_path = ctx.path(_REMOTE_PATCH_DIR).get_child(name)
-    download_info = ctx.download(
+    patch_path = ctx.path(_REMOTE_PATCH_DIR).get_child(str(hash(patch_url)))
```

Comment:
The hash function is of low quality, so how about we concatenate name and hash?

Reply:
fair enough, I also considered `patch_url.replace("/", "_")` but was worried about Windows path lengths...

Reply:
Yeah, those basenames should be unique per repo right? If we just keep them, it'll be a bit easier to debug when the patch doesn't apply? I don't feel strongly against appending a hash to the downloaded file name, but I don't think it's really necessary?

Reply:
Ah, I see, I was confused with how we specify patches in the BCR, this is a more low level feature in http_archive which might have this problem, sorry for the noise.